src/PersonInfoHandler.ts

```diff
@@ -0,0 +1,70 @@
+import { LambdaInterface } from "@aws-lambda-powertools/commons";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { Metrics } from "@aws-lambda-powertools/metrics";
+import { APIGatewayProxyEvent } from "aws-lambda";
+import { PersonInfoRequestProcessor } from "./services/PersonInfoRequestProcessor";
+import { HttpCodesEnum } from "./models/enums/HttpCodesEnum";
+import { MessageCodes } from "./models/enums/MessageCodes";
+import { getParameter } from "./utils/Config";
+import { Constants, EnvironmentVariables } from "./utils/Constants";
+import { AppError } from "./utils/AppError";
+import { checkEnvironmentVariable } from "./utils/EnvironmentVariables";
+import { Response } from "./utils/Response";
+import { getSessionIdHeaderErrors } from "./utils/Validations";
+
+const { POWERTOOLS_METRICS_NAMESPACE = Constants.BAV_METRICS_NAMESPACE, POWERTOOLS_LOG_LEVEL = "DEBUG", POWERTOOLS_SERVICE_NAME = Constants.ABORT_LOGGER_SVC_NAME } = process.env;
+
+export const logger = new Logger({
+	logLevel: POWERTOOLS_LOG_LEVEL,
+	serviceName: POWERTOOLS_SERVICE_NAME,
+});
+
+let PUBLIC_KEY: string;
+
+const metrics = new Metrics({ namespace: POWERTOOLS_METRICS_NAMESPACE, serviceName: POWERTOOLS_SERVICE_NAME });
+
+export class PersonInfoHandler implements LambdaInterface {
+
+	@metrics.logMetrics({ throwOnEmptyMetrics: false, captureColdStartMetric: true })
+
+	async handler(event: APIGatewayProxyEvent, context: any): Promise<Response> {
+		logger.setPersistentLogAttributes({});
+		logger.addContext(context);
+
+		try {
+			const sessionId = this.validateEvent(event);
+			const publicKeyPath = checkEnvironmentVariable(EnvironmentVariables.PUBLIC_KEY_SSM_PATH, logger);
+    	PUBLIC_KEY = await getParameter(publicKeyPath);
```

Comment:
If you add a check for PUBLIC_KEY at this point you won't always need it fetch it cause it'll be in Lambda memory right?

Reply:
Yeah good point, added